experimental/packages/opentelemetry-sdk-metrics-base/src/state/MetricStorageRegistry.ts

```diff
@@ -0,0 +1,93 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { MetricStorage } from './MetricStorage';
+import { isDescriptorCompatibleWith } from '../InstrumentDescriptor';
+import * as api from '@opentelemetry/api';
+import { Maybe } from '../utils';
+import { getConflictResolutionRecipe, getIncompatibilityDetails } from '../view/RegistrationConflicts';
+
+/**
+ * Internal class for storing {@link MetricStorage}
+ */
+export class MetricStorageRegistry {
+  private readonly _metricStorageRegistry = new Map<string, MetricStorage[]>();
+
+  getStorages(): MetricStorage[] {
+    const storages = [];
+    for (const metricStorages of Array.from(this._metricStorageRegistry.values())) {
+      storages.push(...metricStorages);
```

Comment:
I'd prevent using spread syntax when the target is known to be an array. Since the spread syntax is going to iterate over a javascript iterable object of the array, while `concat` is a native function that can access the array elements and copy them to the new location directly, the `concat` could be faster.

```suggestion
      storages = storages.concat(metricStorages);
```

<details>
<summary>Naive microbenchmark (Click to see details)</summary>

```js
const arr = new Array(1000);
arr.fill(1);
const n = 1000000;

console.time('spread');
for (let i = 0; i < n; i++) {
  const result = [];
  result.push(...arr);
}
console.timeEnd('spread');

console.time('concat');
for (let i = 0; i < n; i++) {
  let result = [];
  result = result.concat(arr);
}
console.timeEnd('concat');
```

Result:
```
spread: 2.767s
concat: 635.634ms
```
</details>

Reply:
Thanks for the feedback. I was not aware of this but I'll make sure to look out for it in the future as well! :slightly_smiling_face: 
Changed in [cf680f6](https://github.com/open-telemetry/opentelemetry-js/pull/2734/commits/cf680f61c10ddeb8f407b87c9c3ec46b378f04ba).